File: frontend/src/services/attendanceService.js
import http from './httpClient'

export function clockIn() {
  return http.post('/api/attendance/clock-in')
}

export function clockOut(remark) {
  return http.post('/api/attendance/clock-out', remark ? { remark } : {})
}

export function getTodayRecord() {
  return http.get('/api/attendance/records/today')
}

export function listRecords(params) {
  return http.get('/api/attendance/records', { params })
}

File: backend/src/main/java/com/oclock/checkin/attendance/service/AttendanceService.java
package com.oclock.checkin.attendance.service;

import com.oclock.checkin.attendance.dto.ClockRecordResponse;
import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {

    ClockRecordResponse clockIn(Long memberId);

    ClockRecordResponse clockOut(Long memberId, String remark);

    ClockRecordResponse getTodayRecord(Long memberId);

    List<ClockRecordResponse> listRecords(Long memberId, LocalDate startDate, LocalDate endDate);
}
